graylog2-web-interface/src/components/indices/IndexSummary.jsx

```diff
@@ -9,28 +11,37 @@ const IndexSummary = React.createClass({
     index: React.PropTypes.object.isRequired,
     indexRange: React.PropTypes.object.isRequired,
     isDeflector: React.PropTypes.bool.isRequired,
+    name: React.PropTypes.string.isRequired,
   },
   getInitialState() {
     return { showDetails: this.props.isDeflector };
   },
-  _deflectorBadge() {
-    if (this.props.isDeflector) {
-      return <i className="fa fa-bolt" title="Write-active index"/>;
+  _formatLabels(index) {
+    const labels = [];
+    if (index.is_deflector) {
+      labels.push(<Label key={`${this.props.name}-deflector-label`} bsStyle="primary">deflector</Label>);
     }
-  },
-  _reopenedBadged(index) {
-    if (index.reopened) {
-      return <i className="fa fa-retweet" title="Reopened index"/>;
+
+    if (index.is_closed) {
+      labels.push(<Label key={`${this.props.name}-closed-label`} bsStyle="warning">closed</Label>);
+    }
+
+    if (index.is_reopened) {
+      labels.push(<Label key={`${this.props.name}-reopened-label`} bsStyle="success">reopened</Label>);
     }
+
+    return <span>{labels}</span>;
   },
+
   _formatIndexRange() {
     if (this.props.isDeflector) {
-      return 'Contains messages up to ' + new DateTime().toRelativeString();
+      return `Contains messages up to ${new DateTime().toRelativeString()}`;
     }
 
-    if (this.props.index.all_shards) {
-      const count = this.props.index.all_shards.documents.count;
-      const deleted = this.props.index.all_shards.documents.deleted;
+    const sizes = this.props.index.size;
+    if (sizes) {
+      const count = sizes.count;
```

Comment:
Is this calculation right? It looks like `count` is not part of `IndexSizeSummary`.